src/amp/stores/index.js

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable func-names */
+import { flow, getEnv } from 'mobx-state-tree';
+import base from '../../shared/stores';
+import { home, single } from '../../shared/contexts';
+import { gaVars, gaTriggers } from '../analytics';
+
+export default base.actions(self => ({
+  fetchSelectedItem: flow(function*() {
+    const { connection, settings } = self.root;
+    const selectedItem = getEnv(self).initialSelectedItem;
+
+    if (selectedItem.page) {
+      const { menu } = settings.theme;
+      const context = home(menu);
+      connection.routeChangeSucceed({
+        selectedItem,
+        context,
+      });
+      yield connection.fetchListPage(selectedItem);
+    } else {
+      const context = single();
+      connection.routeChangeSucceed({
+        selectedItem,
+        context,
+      });
+      yield connection.fetchEntity(selectedItem);
+    }
+  }),
+  fetchMenuTaxonomies: flow(function*() {
+    const { settings, connection } = self.root;
+
+    const menu = settings.theme.menu.reduce((result, current) => {
+      if (current.type !== 'link' && current.type !== 'latest') {
+        if (result[current.type])
+          result[current.type].push(current[current.type]);
+        else result[current.type] = [current[current.type]];
+      }
+
+      return result;
+    }, {});
+
+    if (menu.category) {
+      yield connection.fetchCustomPage({
+        name: 'menuCategories',
+        type: 'category',
+        page: 1,
+        params: { include: menu.category.join(','), per_page: 99 },
+      });
+    }
+
+    if (menu.tag) {
+      yield connection.fetchCustomPage({
+        name: 'menuTags',
+        type: 'tags',
+        page: 1,
+        params: { include: menu.tag.join(','), per_page: 99 },
+      });
+    }
+  }),
+  fetchShareCount: flow(function*() {
+    const { type, id } = getEnv(self).initialSelectedItem;
+    yield self.share.all.requestCount({ type, id });
+  }),
+  beforeSsr: flow(function*() {
+    const { settings, analytics, connection } = self.root;
+    self.lang.setLang(settings.theme.lang);
+
+    if (analytics && analytics.googleAnalytics) {
+      analytics.googleAnalytics.setAmpVars(gaVars);
+      analytics.googleAnalytics.setAmpTriggers(gaTriggers);
+    }
+
+    yield self.fetchSelectedItem();
+    yield connection.fetchListPage({ type: 'latest', id: 'post', page: 1 });
```

Comment:
I think this should be done within `fetchSelectedItem` (which can use another name) because if not, we will always be fetching the home, even for category or tag lists.